examples/plot_ECoG_preprocessing_tutorial.py

```diff
@@ -0,0 +1,242 @@
+from scipy.signal.windows import kaiser
+
+# %% 
+#
+
+
+window = kaiser(1024, 2)
+freq, car_pwr = welch(car_data[:,0], fs=new_sample_rate, window=window)
+_, nth_pwr = welch(nth_data[:,0], fs=new_sample_rate, window=window)
+
+fig, axs = plt.subplots(1,2, figsize=(10,4), sharey=True, sharex=True)
+#fig.tight_layout()
+axs[0].semilogy(freq, car_pwr)
+axs[0].set_xlabel('Frequency (Hz)')
+axs[0].set_ylabel('Power density (V^2/Hz)')
+axs[0].set_xlim([1, 150])
+axs[0].set_title('Pre notch filtering')
+
+axs[1].semilogy(freq, nth_pwr)
+axs[1].set_xlabel('Frequency (Hz)')
+axs[1].set_xlim([1, 150])
+axs[1].set_title('Post notch filtering')
+fig.tight_layout()
+pass
+
+
+# %%
+#  Time-frequency decomposition with wavelets
+# ------------------------
+# - Here we decompose the neural time series into 6 different frequency bands using a wavelet transform
+# - The wavelet transform outputs the analytical amplitude, which is complex valued.
+# - I'm plotting the magnitude of analytical amplitude which is a proxy of band power
+
+from process_nwb import wavelet_transform
+
+# %% 
+#
+
+
+tf_data, _, ctr_freq, bw = wavelet_transform.wavelet_transform(nth_data, new_sample_rate, filters='rat', hg_only=True, X_fft_h=None, npad=None)
+
+
+# %%
+#
+
+ctr_freq[:]
+
+# %%
+#
+
+tf_data.shape
+
+# %%
+#
+
+num_tf_signals = len(ctr_freq)
+fig, axs = plt.subplots(num_tf_signals, 1, sharex=True, sharey=True, figsize=(15,10))
+fig.subplots_adjust(hspace=0.4)
+fig.tight_layout
+
+time = np.linspace(0, duration, tf_data.shape[0])
+for idx in range(num_tf_signals):
+    sig = np.abs(tf_data[:,0,idx])
+    axs[idx].plot(time, sig)
+    axs[idx].set_title('Frequency = {0:.0f} ; Bandwidth = {1:0.0f}'.format(ctr_freq[idx], bw[idx]))
+    axs[idx].set_ylabel('Amp. (au)')
+
+# %%
+# Normalizing power by z-scoring
+# ---------------------------
+# - For neural data, power falls off with frequency so it can be difficult to compare amplitude changes across different frequency bands.
+# - To address this issue we normalize each frequency band by zscoring. 
+# - In this case, the mean and standard deviation are calculated over the entire signal, but if there's a baseline period(s) then it would make more sense to calculate mean and standard deviation over those period(s)
+
+def zscore_signal(signal, mean=None, std=None):
+    if mean is None:
+        mean = np.mean(signal)
+    if std is None:
+        std = np.std(signal)
+    return (signal - mean)/std
+
+
+# %%
+#
+
+
+tf_norm_data = np.empty(tf_data.shape, dtype='complex')
+
+for idx in range(num_tf_signals):
+    for ch in range(num_channels):
+        tf_norm_data[:,ch,idx] = zscore_signal(tf_data[:,ch,idx])
+
+
+# %%
+#
+
+
+num_tf_signals = len(ctr_freq)
+fig, axs = plt.subplots(num_tf_signals, 1, sharex=True, sharey=True,figsize=(15,10))
+fig.subplots_adjust(hspace=0.4)
+fig.tight_layout
+
+time = np.linspace(0, duration, tf_data.shape[0])
+for idx in range(num_tf_signals):
+    sig = np.abs(tf_norm_data[:,0,idx])
```

Comment:
Take ABS above.

Reply:
<img width="721" alt="Abs_First" src="https://user-images.githubusercontent.com/11592167/108003737-7864e880-6fa8-11eb-89e3-eb5c4a532be9.png">
<img width="717" alt="Original" src="https://user-images.githubusercontent.com/11592167/108003744-7bf86f80-6fa8-11eb-88f4-00a40ef6d733.png">
Adjusts the Y axis after taking abs first on this graph, might not matter, but I'm not sure

Reply:
Top graph is the one produced after taking abs first, bottom graph is original

Reply:
@hlillemark Jesse is right. We should take ABS above and zscore amplitude, not complex values. Top graph is correct.